### PR TITLE
Support xrefs for type/data constructors and data fields.

### PR DESCRIPTION
**Pull request by jinwoo**
Fixes https://github.com/google/haskell-indexer/issues/6.

### CODE REVIEW

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Compat.hs [hunk 1]
```diff
@@ -420,6 +420,27 @@ pattern IEVarCompat lwn <-
   IEVar _ lwn
 #endif
 
+pattern IEThingAbsCompat lwn <-
+#if __GLASGOW_HASKELL__ < 806
```
Comment (judah) [inline haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Compat.hs]: For what it's worth, I'd support removing everything lower than 8.6 given how significantly the API changed in that version.

Comment (jinwoo) [reply haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Compat.hs]: I filed https://github.com/google/haskell-indexer/issues/98, which is about removing support for 8.0. But yeah, I want to remove all the versions <0.6.

But until that happens, I think we should keep the builds green. Hopefully I can work on that soon once we have a consensus.

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs [hunk 2]
```diff
@@ -470,19 +471,40 @@ refsFromRenamed ctx declAlts (hsGroup, importDecls, _, _) =
...
+    refsFromImport refKind (L _ (IEThingAllCompat (L l n))) =
+      maybeToList $ give ctx (nameLocToRef (ieWrappedName n) refKind l)
+    refsFromImport refKind (L _ (IEThingWithCompat (L tl tn) ctors fields)) =
+      typeRef ++ ctorRefs ++ fieldRefs
+      where
+        typeRef =
+          maybeToList $ give ctx (nameLocToRef (ieWrappedName tn) refKind tl)
+        ctorRefs =
+          catMaybes
+            $ map
```
Comment (robinp) [inline haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs]: nit: `catMaybes . map = mapMaybe` ?

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Compat.hs (hunk 1 above)
Comment (jinwoo) [reply haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Compat.hs]: All right. I changed the title of #98 so repurpose it to removing versions <8.6 and supporting 8.8.

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs (hunk 2 above)
Comment (jinwoo) [reply haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs]: Yup.

**Pull request merged**